asyncpg/pool.py

```diff
@@ -201,7 +201,8 @@ def __init__(self, pool, *, connect_args, connect_kwargs,
                     await asyncio.wait_for(
                         self._con._protocol._wait_for_cancellation(),
                         timeout, loop=self._pool._loop)
-                    budget -= time.monotonic() - started
+                    if budget is not None:
```

Comment:
Can you add a test that would fail without this change?

Reply:
Done.